### ISSUE TITLE
still possible to run?

### ISSUE THREAD

**Issue by saberworks** [microcosm-cc/export-vbulletin#2]
I was just writing this exact same thing in perl and decided to search github halfway through.  Is there any way to run this nowadays?  I followed the instructions as best I could.  I'm on ubuntu 20.04 which go version 1.13.8 installed. 

In the Makefile, there's a line that says `go tool vet ...` which I had to change to `go vet ...` because `go tool vet` errors out, no longer supported, replaced by `go vet`.

After that, I type `make` and it errors out:

```
gotest@oree:~/go/src/github.com/microcosm-cc/export-vbulletin$ make
can't load package: package export: cannot find package "export" in any of:
	/usr/lib/go-1.13/src/export (from $GOROOT)
	/home/gotest/go/src/export (from $GOPATH)
make: *** [Makefile:31: vet] Error 1
```

Google searches indicate there should be a file called `export.go` in the `export/` folder, which there is.  Any suggestions on fixing it?

```
gotest@oree:~/go/src/github.com/microcosm-cc/export-vbulletin$ echo $GOPATH
/home/gotest/go
gotest@oree:~/go/src/github.com/microcosm-cc/export-vbulletin$ echo $GOROOT

gotest@oree:~/go/src/github.com/microcosm-cc/export-vbulletin$
```

Thank you!

**Comment by buro9** [comment]
Ugh, the code will work but the code was written 9+ years ago and so it knows nothing of the go mod build system that was introduced long after.

I'll see what I can do for you.

**Comment by saberworks** [comment]
Please don’t spend too much time on this. I wrote an exporter in perl that’s doing everything except attachments and those don’t look too difficult anyway. I don’t know anything about go and was just thinking there might be a trivial fix.

**Comment by buro9** [comment]
Just updated it... the code should be readable and make sense.

It's not perfect and didn't try to be, the major limitations are documented in the README.

Re-open if you can't run it... I mean, it's been almost a decade, I've got it building easily again but I don't have things to test it against as I exported a decade ago :D

**Comment by buro9** [comment]
PS: Assumption is that it's being run on Linux... if it isn't, remove line 24 and 25 of the Makefile.

My use-case back then was that it was built on Mac but run on Linux. But if you're running it on the same system then you don't need to build for Linux specifically.

**Comment by saberworks** [comment]
Thanks for the modifications!  I was able to get it to run... to a certain point.  I had to make some changes (patch attached) because the `vb_` prefix was hard coded in a few places.

The `cis_thread_ignore` table doesn't exist on my database (comments indicated it was from a plugin or something), so I just returned prior to hitting that (not included in the patch).

Unfortunately I found that my database / forum version must be newer than yours.  Only now I realize (found this Issue filed: #1 ) that this is for 3.x and I have 4.x.  I am amazed that it got through as much as it did!  It is failing to export the attachments:

```
Failed on ID 4739 : Error 1054 (42S22): Unknown column 'a.postid' in 'field list'
```

It looks like in my version the attachments aren't directly linked to posts, because attachments can be either added to posts or "Albums".  I haven't yet figured out how to link posts to their attachments otherwise I'd send a better patch.

I will probably go back to using the perl version I started just because I've never used go before and I actually got up to this point also (figuring out attachments).  Thank you!!

[export-vbulletin-patch.txt](https://github.com/microcosm-cc/export-vbulletin/files/11413238/export-vbulletin-patch.txt)
